layers.go

```diff
@@ -970,6 +979,77 @@ func (r *layerStore) SetNames(id string, names []string) error {
 	return ErrLayerUnknown
 }
 
+func (r *layerStore) datadir(id string) string {
+	return filepath.Join(r.layerdir, id)
+}
+
+func (r *layerStore) datapath(id, key string) string {
+	return filepath.Join(r.datadir(id), makeBigDataBaseName(key))
+}
+
+func (r *layerStore) BigData(id, key string) (io.ReadCloser, error) {
+	if key == "" {
+		return nil, errors.Wrapf(ErrInvalidBigDataName, "can't retrieve layer big data value for empty name")
+	}
+	layer, ok := r.lookup(id)
+	if !ok {
+		return nil, errors.Wrapf(ErrLayerUnknown, "error locating layer with ID %q", id)
+	}
+	return os.Open(r.datapath(layer.ID, key))
```

Comment:
Note, to be explicit: actual reads happen outside of the layer lock. AFAICS this is safe because removing the file does not truncate the underlying inode, and writes happen via `NewAtomicFileWriter`, i.e. don’t overwrite the original.

(Perhaps add comment to the `NewAtomicFileWriter` call below to note that the read implementation relies on it?)

Reply:
thanks.

Added a comment